assets/php/class-wgpb-block-library.php

```diff
@@ -0,0 +1,254 @@
+	 * @param array  $deps   Optional. An array of registered stylesheet handles this stylesheet depends on. Default empty array.
+	 * @param string $media  Optional. The media for which this stylesheet has been defined. Default 'all'. Accepts media types like
+	 *                       'all', 'print' and 'screen', or media queries like '(orientation: portrait)' and '(max-width: 640px)'.
+	 */
+	protected static function register_style( $handle, $src, $deps = array(), $media = 'all' ) {
+		$filename = str_replace( plugins_url( '/', WGPB_PLUGIN_FILE ), '', $src );
+		$ver      = self::get_file_version( $filename );
+		wp_register_style( $handle, $src, $deps, $ver, $media );
+	}
+
+	/**
+	 * Register block scripts & styles.
+	 */
+	public static function register_assets() {
+		self::register_style( 'wc-block-editor', plugins_url( 'build/editor.css', WGPB_PLUGIN_FILE ), array( 'wp-edit-blocks' ) );
+		self::register_style( 'wc-block-style', plugins_url( 'build/style.css', WGPB_PLUGIN_FILE ), array() );
+
+		// Shared libraries and components across all blocks.
+		self::register_script( 'wc-vendors', plugins_url( 'build/vendors.js', WGPB_PLUGIN_FILE ), array(), false );
+
+		$block_dependencies = array(
+			'wp-api-fetch',
+			'wp-blocks',
+			'wp-components',
+			'wp-compose',
+			'wp-data',
+			'wp-element',
+			'wp-editor',
+			'wp-i18n',
+			'wp-url',
+			'lodash',
+			'wc-vendors',
+		);
+
+		self::register_script( 'wc-handpicked-products', plugins_url( 'build/handpicked-products.js', WGPB_PLUGIN_FILE ), $block_dependencies );
+		self::register_script( 'wc-product-best-sellers', plugins_url( 'build/product-best-sellers.js', WGPB_PLUGIN_FILE ), $block_dependencies );
+		self::register_script( 'wc-product-category', plugins_url( 'build/product-category.js', WGPB_PLUGIN_FILE ), $block_dependencies );
+		self::register_script( 'wc-product-new', plugins_url( 'build/product-new.js', WGPB_PLUGIN_FILE ), $block_dependencies );
+		self::register_script( 'wc-product-on-sale', plugins_url( 'build/product-on-sale.js', WGPB_PLUGIN_FILE ), $block_dependencies );
+		self::register_script( 'wc-product-top-rated', plugins_url( 'build/product-top-rated.js', WGPB_PLUGIN_FILE ), $block_dependencies );
+		self::register_script( 'wc-products-attribute', plugins_url( 'build/products-attribute.js', WGPB_PLUGIN_FILE ), $block_dependencies );
+		self::register_script( 'wc-featured-product', plugins_url( 'build/featured-product.js', WGPB_PLUGIN_FILE ), $block_dependencies );
+	}
+
+	/**
+	 * Register blocks, hooking up assets and render functions as needed.
+	 */
+	public static function register_blocks() {
+		require_once dirname( __FILE__ ) . '/class-wgpb-block-featured-product.php';
+
+		register_block_type(
+			'woocommerce/handpicked-products',
+			array(
+				'editor_script' => 'wc-handpicked-products',
+				'editor_style'  => 'wc-block-editor',
+				'style'         => 'wc-block-style',
```

Comment:
Minor/optional and doesn't need addressed in this PR (since this code is just being moved), but if these are always going to have the same, maybe we could register a helper to prevent some of the duplication?

Reply:
I was thinking about that, but i'm not sure if these will _always_ be shared. For the immediate/mid-term future, probably, but I'm not sure if it's worth creating another helper function.